src/model_config_utils.cc

```diff
@@ -784,6 +784,28 @@ GetNormalizedModelConfig(
   return Status::Success;
 }
 
+Status
+SetDefaultInstanceCount(
+    ::inference::ModelInstanceGroup* group, const std::string& backend)
+{
+  const int default_cpu_instance_count = 2;
+  if (group->kind() == inference::ModelInstanceGroup::KIND_CPU) {
+    // Some backends don't perform well with multiple
```

Comment:
I think it might be better to only set the number of instances to two for backends we know than turn it on for all the backends. For Python backend, if the model is using Pytorch library, it would add ~600MB of GPU memory for each new model instance. Also, I think it is best to not add backend-specific code in the core repository. Perhaps we can add a backend API so that the backend can indicate the instance kind and instance count that it prefers if the user has not specified anything?

Reply:
For now I have changed the list to opt in rather than opt out to show what it will look like.

Reply:
Thanks for the response. I just wanted to bring-up this change since it keeps coming up. This backend API change will both fix Python backend + jetson issue and remove the backend specific changes from core. I understand that it is easier to hardcode backend names for now. Just something to consider in the future.